### DISCUSSION TITLE
SSR with pre-rendering

### DISCUSSION THREAD

**Discussion by thejspro** [nuxt/nuxt#23481] [Questions]
Just wanted to clarify what happens when I build in SSR mode but also add a few routes to the `prerender` section in the Nuxt config?

My whole website requires SSR, however there are a couple of pages with a lot of content and I would like to generate them statically if possible. I also have a single server-side route which should be static (`/server/api/our-locations`).

This is my `Nuxt.config.js`:

```
nitro: {
    prerender: {
      routes: ["/server/api/our-locations", "/brands", "/sitemap"],
    },
  },
```

Am I right assuming that the entire website will be built for SSR, but the routes above will be pre-rendered at build time and therefore they will be static? What will happen when they are accessed in the browser? Will Nuxt server their pre-rendered static versions or will they be generated on the fly server side?

**Comment by SultanDino** [comment]
Same question

**Comment by TheAlexLichter** [comment]
@goodpixels yes, correct. prerender => "beforehand", static routeRule => "on the fly" but cached until the next build.
